Dw/prov mgmt scripts

dolled-possum opened the pull request:
## Description

This PR adds three terminal scripts for managing the provider config: get-providers, add-provider, and remove-provider.  They are included in terminal help and their usage guidelines are included when arguments aren't supplied. Testing is easy because you can see the results of add- or remove- by looking at the results of get-providers afterward.

dolled-possum committed:
get and remove…

dolled-possum committed:
…add-provider first iteration

dolled-possum committed:
add-provider second

dolled-possum committed:
removing debug logging

dolled-possum committed:
style and text fixes

dolled-possum committed:
final touches on provider management scripts

github-actions[bot] committed:
Format Rust code using rustfmt

dolled-possum committed:
Make provider serialization consistent w/init load

dolled-possum committed:
Merge branch 'dw/prov-mgmt-scripts' of https://github.com/hyperware-ai/hyperdrive into dw/prov-mgmt-scripts

dolled-possum committed:
disabling adding node providers

dolled-possum committed:
splits add-provider into node and rpcurl flavors

dolled-possum committed:
better handling of unexpected add responses

dolled-possum committed:
workflow call dynamically uses blue/green

dolled-possum committed:
Merge branch 'develop' into dw/prov-mgmt-scripts

dolled-possum committed:
Merge branch 'develop' into dw/prov-mgmt-scripts

nick1udwig commented on hyperdrive/packages/terminal/add-node-provider/Cargo.toml:
```diff
@@ -0,0 +1,20 @@
+rmp-serde = "1.1.2"
+serde = { version = "1.0", features = ["derive"] }
+wit-bindgen = "0.42.1"
+serde_json = "1.0.140"
```
prefer alphabetical order when there isn't some other logical ordering to use (here and elsewhere)

nick1udwig commented on hyperdrive/packages/terminal/get-providers/src/lib.rs:
```diff
@@ -0,0 +1,87 @@
+    //    Err(e) => {
+    //        format!("Failed to parse response as JSON: {}\nRaw response: {}", e, String::from_utf8_lossy(&body))
+    //    }
+    //}
```
remove old commented-out code

nick1udwig reviewed (approved):
A couple small comments -- merge when ready.

Nice work, sorry it took so long to review! :rocket:

dolled-possum committed:
Merge branch 'develop' into dw/prov-mgmt-scripts

dolled-possum committed:
small fixes

alphabetical order in Cargo.toml
adding new packages to top level workspace members
removing old commented code

github-actions[bot] committed:
Format Rust code using rustfmt
